rust/integration-tests/src/cli_parser.rs

```diff
@@ -0,0 +1,60 @@
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+// Define a global static to store the parsed arguments
+#[allow(dead_code)]
+static TEST_CONFIG: Lazy<Mutex<TestArgs>> = Lazy::new(|| {
+    let args = parse_test_args();
+    Mutex::new(args)
+});
+
+// function to fetch global test args
+#[allow(dead_code)]
+pub fn get_test_config() -> (bool, Option<String>) {
+    let test_args = TEST_CONFIG.lock().unwrap().clone();
+    (test_args.generate_output, test_args.output_path)
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct TestArgs {
+    pub generate_output: bool,
+    pub output_path: Option<String>,
+}
+
+#[allow(dead_code)]
+pub fn parse_test_args() -> TestArgs {
```

Comment:
does this helpfully complain if insufficient/bad args are provided?

Reply:
one thing is that we would have to passing addition `--` to bypass failing unrecognized argument from cargo test

```
1. cargo test
2. cargo test diff_tests
3. cargo test diff_tests -- --nocapture --generate-output
4. cargo test diff_tests -- -- --generate-output <-- like this 
5. cargo test -- --nocapture
```